### ISSUE TITLE
Problems using HDF5Matrix

### ISSUE THREAD

**Issue by singlasahil14** [fchollet/keras#5307]
https://gist.github.com/singlasahil14/b153efaea6549f7dacff91e27b7c356b
The above is a gist of my script I am using to train my model using an HDF5Matrix. I have precomputed convolutional features. I need to train fully connected layers of a VGG net on them. 

This is the important part of the script.
<img width="633" alt="screen shot 2017-02-07 at 6 56 54 pm" src="https://cloud.githubusercontent.com/assets/6660192/22692961/5003216c-ed67-11e6-83fd-7ee46313ef31.png">

The following is the result:
<img width="773" alt="screen shot 2017-02-07 at 6 07 16 pm" src="https://cloud.githubusercontent.com/assets/6660192/22691396/4f65b8a2-ed60-11e6-8179-230f69dd4b39.png">

The first time, model is fetching data from HDF5Matrix. The second time from a raw numpy array. Both the models are trained from scratch. You can see how slowly the training proceeds for the first case. (0.2963 after 4 epochs). In the second case, it gives 0.9970 after 4 epochs.

**Comment by singlasahil14** [comment]
Okay, so the issue here is that the pretrained convolutional features were not shuffled. So the data was initially in a not shuffled order (data of same class clustering together). So every batch will contain the same classes in most cases. Batch shuffle just shuffles the order in which different batches are seen by the epoch. But the data in each batch remains the same in every epoch. 

I shuffled the data when convolutional features are calculated and this got resolved.
